oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java

```diff
@@ -869,6 +872,11 @@ public int getMemory() {
         }
     }
 
+
+    public boolean isChildOrderCleanupEnabled() {
+        return noChildOrderCleanupFeature == null || !noChildOrderCleanupFeature.isEnabled();
```

Comment:
Having second thought on the default being `true` .. we should play it more safe by using `false` but aim to change it to `true` once we have seen it _behave_ well in productive systems?

Reply:
Hmmm; in that case we'd need to either disable the new tests, or change the toggle just for those. Which would mean that we'd be testing something that wouldn't be enabled by default. That sounds likes asking for trouble.

This is a bug fix, not a ew feature, and the toggle really is a safety net in case something is wrong. Thus I would leave the default set to true.

Reply:
deadlock nr 2 ;)

Reply:
I'll repeat what I said already. Do not overthink this. This is a bug fix. And we have a safety net in case it causes trouble. How certain downstream projects ship this change to their customers should not affect how we handle this in this open-source project.